### ISSUE TITLE
Rate Limiting After Authentication

### ISSUE THREAD

**Issue by osmansonmez** [ThreeMammals/Ocelot#662]
## Steps to Reproduce the Problem
I want to use RateLimiting but i want to take requestid from claims. I see RateLimiting  handler runs before authentication handler and ClaimsToHeadersMiddleware handler.   I think we should change order RateLimiting and authentication . I dont trust clear haeder value. Client can change clientid value always and make request to server and can break ratelimiting control. I want to take requestID from in to the claims after authentication.  How can we do?

  1.
  1.
  1.

## Specifications

  - Version:
  - Platform:
  - Subsystem:

**Comment by TomPallister** [comment]
@osmansonmez yes this makes sense I think, I would accept a PR for this but don't have time to do the work myself atm. I will mark this issue as a new feature.

**Comment by raman-m** [comment]
@harindaka commented [on Sep 12, 2019](#issuecomment-530743183):

> this can be exploited to launch a dos attack on a valid client in its current state

Wotta?

----
> mark as experimental maybe?

Can I mark your message as experimental? 😉
